cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/streaming/function/StreamingBatchSinkFunction.java

```diff
@@ -55,11 +58,15 @@
   private final JavaSparkExecutionContext sec;
   private final StageSpec stageSpec;
   private final FunctionCache functionCache;
+  @Nullable
+  private final DatasetContext datasetContext;
 
-  public StreamingBatchSinkFunction(JavaSparkExecutionContext sec, StageSpec stageSpec, FunctionCache functionCache) {
+  public StreamingBatchSinkFunction(JavaSparkExecutionContext sec, StageSpec stageSpec, FunctionCache functionCache,
+                                    @Nullable DatasetContext datasetContext) {
```

Comment:
i'm not clear on why we need to pass in the dataset context, what was wrong with the way it was doing it before with the sec starting a new transaction for prepareRun() and onRunFinish()?

Reply:
This was done because of getting an error related to nested transactions. After addressing the above comment regarding executing sinks out of the transaction, this will not be required, so reverted those changes.